jaeger-core/src/main/java/com/uber/jaeger/senders/HttpSender.java

```diff
@@ -90,29 +86,28 @@ public HttpSender(String endpoint, int maxPacketSize, OkHttpClient client) {
   }
 
   private HttpSender(Builder builder) {
-    super(new TBinaryProtocol.Factory(), builder.maxPacketSize);
+    super(ProtocolType.Binary, builder.maxPacketSize);
     HttpUrl collectorUrl = HttpUrl
         .parse(String.format("%s?%s", builder.endpoint, HTTP_COLLECTOR_JAEGER_THRIFT_FORMAT_PARAM));
     if (collectorUrl == null) {
       throw new IllegalArgumentException("Could not parse url.");
     }
     this.httpClient = builder.clientBuilder.build();
     this.requestBuilder = new Request.Builder().url(collectorUrl);
-    this.serializer = new TSerializer(protocolFactory);
   }
 
   @Override
-  public void send(Process process, List<Span> spans) throws TException {
+  public void send(Process process, List<Span> spans) throws Exception {
```

Comment:
Is it possible to have a more specialized exception?

Reply:
Actually I think it could be changed to `SenderException` - will give it a go.